### PR TITLE
Refactor playback mapping

### PR DESCRIPTION
**Pull request by ladanazita**

### CODE REVIEW

File: Pod/Classes/SEGAdobeIntegration.m [hunk 1]
```diff
@@ -425,6 +397,23 @@ - (void)trackHeartbeatEvents:(SEGTrackPayload *)payload
...
     }
+
+    NSDictionary *videoTrackEvents = @{
+        @"Video Playback Buffer Started" : @(ADBMediaHeartbeatEventBufferStart),
+        @"Video Playback Buffer Completed" : @(ADBMediaHeartbeatEventBufferComplete),
+        @"Video Playback Seek Started" : @(ADBMediaHeartbeatEventSeekStart),
+        @"Video Playback Seek Completed" : @(ADBMediaHeartbeatEventSeekComplete)
+    };
+
+    for (NSString *event in videoTrackEvents) {
```
Comment (f2prateek) [inline Pod/Classes/SEGAdobeIntegration.m]: Iterating over the entire dictionary is inefficient. We should remove it - the check should just check if the event exists in the `videoTrackEvents` dictionary or not.

Comment (ladanazita) [reply Pod/Classes/SEGAdobeIntegration.m]: Hmm but I want to dynamically set it with the Adobe value

Comment (f2prateek) [reply Pod/Classes/SEGAdobeIntegration.m]: you can still do that.

```
if (videoTrackEvents[event]) {
  enum ADBMediaHeartbeatEvent videoEvent = [videoTrackEvents[event] intValue];
  ...
}
```

Or even:
```
enum ADBMediaHeartbeatEvent videoEvent = [videoTrackEvents[event] intValue];
if videoEvent != 0 {
  ...
}

```

**Comment by ladanazita** [comment]
@f2prateek lmk if this update looks gouda

**Pull request merged**